Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InteractableOnFocus.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// Adds or removes materials to target renderer for highlighting Focused <see href="https://docs.unity3d.com/ScriptReference/GameObject.html">GameObject</see>s.
+    /// </summary>
+    /// <remarks>Useful with focusable <see href="https://docs.unity3d.com/ScriptReference/GameObject.html">GameObject</see>s</remarks>
+    public class InteractableOnFocus : BaseFocusHandler
```

Comment:
This is really cool and having a remote profile on other objects is something I've though about doing for a while.

Some warnings from rendering the inspector for InteractableOnFocus.
`Unexpected top level layout group! Missing GUILayout.EndScrollView/EndVertical/EndHorizontal?
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)`

Two things to consider,
1. It will be interesting to see any other use cases for this pattern, like InteractableOnPress? Do we want to make a base version of this that is more generic? or am I getting ahead of things?
2. When a theme is created the DefaultInteractableStates are added to the theme, which include Default, Focus, Press and Disabled. In this control we only care about default and focus, do we want to only show the default and focus values instead of all 4? There's a way to filter which states show up and which ones will need to be added if they do not exist already. This could be done later, without breaking anything, if we should wait to get some more feedback.

For reference.

![Example of created theme with state added at the top](https://user-images.githubusercontent.com/22085553/62385647-84926d80-b50a-11e9-96c0-b8923b63b305.png)

Also, very cool that the MRTK Standard shader is referenced by default for new themes!

Reply:
I'll look into the warnings 

1) This is sort of the goal I think. I think it will only be so fruitful to break out specialties of the main Interactable class. The true win though is making the InteractableThemes extensible. I could as a developer create whatever kind of input handler model I want but then have extensible visual configurations and not have to hardcode anything in my component (like activate this or that). The key for this though is to actually make the InteractableTheme, just a Theme and not dependent on Interactable. It should almost be a separate feature. 

2) Yes I thought about this...but I'm hesitant to do anything atm. I don't want to filter fields because then people might be weirded out why their asset is only half defined. And then for the InteractableOnFocus to force a certain state object kind of like Interactable does is not the direction I want to push forward either...or at least just yet.